server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -773,7 +777,14 @@ static ClusterState clusterStateCreateIndex(ClusterState currentState, Set<Clust
     static IndexMetadata buildIndexMetadata(String indexName, List<AliasMetadata> aliases,
                                             Supplier<DocumentMapper> documentMapperSupplier, Settings indexSettings, int routingNumShards,
                                             @Nullable IndexMetadata sourceMetadata) {
+        return buildIndexMetadata(indexName, aliases, documentMapperSupplier, indexSettings, routingNumShards, sourceMetadata, false);
+    }
+
```

Comment:
It looks like this version of this method is just used in a single test, so I think we should delete this version of this method and just add the extra `false` to the call in `testBuildIndexMetadata()`